cli/lsp/config.rs

```diff
@@ -15,19 +15,57 @@ pub struct ClientCapabilities {
   pub workspace_did_change_watched_files: bool,
 }
 
+#[derive(Debug, Clone, Default, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CodeLensSettings {
+  #[serde(default)]
+  pub references: bool,
+  #[serde(default)]
+  pub references_all_functions: bool,
+}
+
 #[derive(Debug, Clone, Default, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct WorkspaceSettings {
   pub enable: bool,
   pub config: Option<String>,
   pub import_map: Option<String>,
+  pub code_lens: Option<CodeLensSettings>,
 
   #[serde(default)]
   pub lint: bool,
   #[serde(default)]
   pub unstable: bool,
 }
 
+impl WorkspaceSettings {
+  /// Determine if any code lenses are enabled at all.  This allows short
+  /// circuiting when there are no code lenses enabled.
+  pub fn enabled_code_lens(&self) -> bool {
+    if let Some(code_lens) = &self.code_lens {
+      code_lens.references
```

Comment:
Is this meant to be `code_lens.references && code_lens.references_all_functions` ?

Reply:
No, because `code_lens.references_all_functions` is optional feature enhancement of `code_lens.references` and `code_lens.references` can be `true` while `code_lens.references_all_functions` can be `false`, but `code_lens.references_all_functions` requires `code_lens.references` to be `true` to have any effect.